src/runtime_src/core/edge/skd/xrt_skd.cpp

```diff
@@ -382,14 +444,29 @@ namespace xrt {
 
   // Respond to SCU subdevice PS kernel initialization is done
   void skd::report_ready() {
-    xclSKReport(devHdl,cu_idx,XRT_SCU_STATE_READY);
+    xclSKReport(m_devhdl,m_cu_idx,XRT_SCU_STATE_READY);
   }
 
   void skd::report_fini() {
-    xclSKReport(devHdl,cu_idx,XRT_SCU_STATE_FINI);
+    xclSKReport(m_devhdl,m_cu_idx,XRT_SCU_STATE_FINI);
   }
 
   void skd::report_crash() {
-    xclSKReport(devHdl,cu_idx,XRT_SCU_STATE_CRASH);
+    xclSKReport(m_devhdl,m_cu_idx,XRT_SCU_STATE_CRASH);
   }
+
+  int skd::get_return_offset(std::vector<xrt_core::xclbin::kernel_argument> args) {
+    int return_offset = 1;
+
+    // Calculate offset to write return code into
+    // If the last argument is a global which means there will be 64-bit address and 64-bit size for total of 16 bytes
+    // Else the last argument size will be either 4-bytes or 8 bytes since arguments are 32-bit aligned
+    if(args.back().type == xrt_core::xclbin::kernel_argument::argtype::global)
```

Comment:
If you look at how the return_offset is being calculated between these two formulas, the only thing that is different between the two is:
1. 16
2. (args.back(0.size>4)?8:4

Why don't you calculate this value first and then use it in the return_offset calculation.  Doing so would most likely make it easier to read and support.